packages/reflection/src/TypeReflector.php

```diff
@@ -53,12 +53,15 @@
 
     private bool $isNullable;
 
+    public bool $isVariadic;
+
     public function __construct(
         private PHPReflector|PHPReflectionType|string $reflector,
     ) {
         $this->definition = $this->resolveDefinition($this->reflector);
         $this->isNullable = $this->resolveIsNullable($this->reflector);
         $this->cleanDefinition = str_replace('?', '', $this->definition);
+        $this->isVariadic = $this->reflector instanceof PHPReflectionParameter && $this->reflector->isVariadic();
```

Comment:
I'm not sure if this should go in `TypeReflector`, maybe `ParameterReflector` is a better place? I think you did it this way because the `TypeReflector` was already available in `ConsoleArgumentDefinition` ? But you also have access to `ParameterReflector` from within the `ConsoleArgumentDefinition::fromParameter()` function. 

If that's not sufficient, I think a better approach is to refactor `ConsoleArgumentDefinition` a bit to always store the `ParameterReflector` as well. Seems cleaner than having `isVariadic` on the type.